### PR TITLE
Remove lint exceptions and fix remaining issues

### PR DESCRIPTION
**Pull request by silaselisha**
- [x] I have removed lint exception by commenting it out
- [x] Fixed all remaining issues regarding `golangci-lint` exception

### CODE REVIEW

**Comment by silaselisha** [comment]
@roger2hk in the previous PR created to solve issues #1438 I realized it was failing as a result of a file of type `*os.File` getting closed twice and that's in file `loglist_refresher_test.go` on function `createTempFile`.

**Comment by AlCutter** [comment]
Thanks @silaselisha, I think we might want to "downgrade" some of these exits/panics - I think it's fine to do that in tests, but particularly in longer-lived server processes or client libraries it's likely better to just log these.

**Review by roger2hk — COMMENTED** [review]
Error strings should not be capitalized (unless beginning with an exported name, a proper noun or an acronym) and should not end with punctuation. This is because error strings usually appear within other context before being printed to the user.

https://google.github.io/styleguide/go/decisions#error-strings

**Comment by silaselisha** [comment]
@AlCutter will it be fine for the recommended "downgrading" changes being part of this PR?

**Comment by roger2hk** [comment]
> @roger2hk I'll look into it A.S.P. and also should I make changes to pieces of code that I didn't introduce but broke this rule https://google.github.io/styleguide/go/decisions#error-strings?

You should scope it to the lint changes. If you find some other error strings and would like to make changes, please open a separate pull request to address them.

**Comment by AlCutter** [comment]
> @AlCutter will it be fine for the recommended "downgrading" changes being part of this PR?

@silaselisha yes please

**Comment by silaselisha** [comment]
@AlCutter Thank you, and I'm open for more contribution.

File: trillian/integration/ct_hammer/main.go [hunk 1]
```diff
@@ -256,7 +256,9 @@ func main() {
 		if _, err := io.Copy(os.Stdout, r); err != nil {
 			klog.Exitf("Failed to print banner!")
 		}
-		r.Close()
+		if err := r.Close(); err != nil {
+			klog.Exitf("Failed to close reader %v", err)
```
Comment (roger2hk) [inline trillian/integration/ct_hammer/main.go:260]: nit: missing `:`.

File: trillian/integration/ct_integration.go [hunk 2]
```diff
@@ -911,7 +911,12 @@ func (ls *logStats) fromServer(ctx context.Context, servers string) (*logStats,
 		if err != nil {
 			return nil, fmt.Errorf("getting stats failed: %v", err)
 		}
-		defer httpRsp.Body.Close()
+		defer func() {
+			if err := httpRsp.Body.Close(); err != nil {
+				fmt.Printf("Operation to close http response body failed:  %v\n", err)
```
Comment (roger2hk) [inline trillian/integration/ct_integration.go:916]: nit: remove extra ` `.

**Review by roger2hk — APPROVED** [review]

**Pull request merged**